packages/runtime/runtime-utils/src/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -322,19 +326,21 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
             }
 
             // Update this node to the same GC state it was when the ack corresponding to this summary was processed.
-            this.gcData = gcDetails.gcData !== undefined ? cloneGCData(gcDetails.gcData) : undefined;
-            this.referenceUsedRoutes = gcDetails.usedRoutes !== undefined ? Array.from(gcDetails.usedRoutes) : undefined;
+            this.gcData = gcDetails?.gcData !== undefined ? cloneGCData(gcDetails.gcData) : undefined;
+            this.referenceUsedRoutes = gcDetails?.usedRoutes !== undefined ? Array.from(gcDetails.usedRoutes) : undefined;
             // If there are no used routes in the GC details, set it to have self route which will make the node
             // referenced. This scenario can only happen if the snapshot is from a client where GC was not run or
             // disabled. In both the cases, the node should be referenced.
-            this.usedRoutes = gcDetails.usedRoutes !== undefined ? Array.from(gcDetails.usedRoutes) : [""];
-
-            // Generate the GC data and used routes of children GC nodes and add it to their snapshot tree.
-            const gcDetailsMap = unpackChildNodesGCDetails(gcDetails);
-            const { childrenTree } = parseSummaryForSubtrees(snapshotTree);
-            gcDetailsMap.forEach((childGCDetails: IGarbageCollectionDetailsBase, childId: string) => {
-                childrenTree.trees[childId].blobs[gcTreeKey] = JSON.stringify(childGCDetails);
-            });
+            this.usedRoutes = gcDetails?.usedRoutes !== undefined ? Array.from(gcDetails.usedRoutes) : [""];
+
+            if (gcDetails !== undefined) {
```

Comment:
Consider `if (gcDetails === undefined) { return; }`

Reply:
Moved the GC specific functionality to a separate function and updated it.